mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -162,23 +165,69 @@ public void setFlags(Flags newState, MessageManager.FlagsUpdateMode replace, Mes
     }
 
     @Override
-    public void delete(MessageId messageId, List<MailboxId> mailboxIds, MailboxSession mailboxSession) throws MailboxException {
+    public DeleteResult delete(MessageId messageId, List<MailboxId> mailboxIds, MailboxSession mailboxSession) throws MailboxException {
         MessageIdMapper messageIdMapper = mailboxSessionMapperFactory.getMessageIdMapper(mailboxSession);
-        MailboxMapper mailboxMapper = mailboxSessionMapperFactory.getMailboxMapper(mailboxSession);
 
         assertRightsOnMailboxes(mailboxIds, mailboxSession, Right.DeleteMessages);
 
-        ImmutableList<MetadataWithMailboxId> metadataWithMailbox = messageIdMapper
+        List<MailboxMessage> messageList = messageIdMapper
             .find(ImmutableList.of(messageId), MessageMapper.FetchType.Metadata)
             .stream()
             .filter(inMailboxes(mailboxIds))
+            .collect(Guavate.toImmutableList());
+
+        if (!messageList.isEmpty()) {
+            delete(messageIdMapper, messageList, mailboxIds, mailboxSession);
+            return DeleteResult.builder()
```

Comment:
you should add static methods for simple cases like `DeleteResult.destroyed(messageId)` or `DeteleResult.notFound(messageId)`